go/mysql/conn_params.go

```diff
@@ -28,15 +30,19 @@ type ConnParams struct {
 	Flags      uint64 `json:"flags"`
 	Flavor     string `json:"flavor,omitempty"`
 
-	// The following SSL flags are only used when flags |= 2048
-	// is set (CapabilityClientSSL).
-	SslCa            string `json:"ssl_ca"`
-	SslCaPath        string `json:"ssl_ca_path"`
-	SslCert          string `json:"ssl_cert"`
-	SslKey           string `json:"ssl_key"`
-	TLSMinVersion    string `json:"tls_min_version"`
-	ServerName       string `json:"server_name"`
-	ConnectTimeoutMs uint64 `json:"connect_timeout_ms"`
+	// The following SSL flags control the SSL behavior.
+	//
+	// Not setting this value implies preferred mode unless
+	// the CapabilityClientSSL bit is set in db_flags. In the
+	// flag is set, it ends up equivalent to verify_identity mode.
+	SslMode          vttls.SslMode `json:"ssl_mode"`
+	SslCa            string        `json:"ssl_ca"`
+	SslCaPath        string        `json:"ssl_ca_path"`
+	SslCert          string        `json:"ssl_cert"`
+	SslKey           string        `json:"ssl_key"`
```

Comment:
Nit: what do you think if we make the `Ssl` prefix all uppercase `SSL`?

Reply:
@fatih I think in Go only TLS gets capitalized like that by default. So if I'd update these, the linter will complain about them which is why I kept them as is (and we use this capitalization in other places too).

Reply:
Gotcha, that makes sense 👍🏼